### PR TITLE
created table copmponent

### PR DESCRIPTION
**Pull request by Ran-Regev-HLS**
I have:
- [ ] deleted all irrelevant comments.
- [ ] deleted all irrelevant logs and debuggers.
- [ ] deleted all sensitive information.
- [ ] made sure my code is generic and reusable for future things or reused existing resources.
- [ ] made sure that my code is readable.
- [ ] checked that the functions and resource names are self-explanatory.
- [ ] tested the code locally and it works as expected.
- [ ] added validations and error handling in relevant cases (external requests, integrations, etc).

### CODE REVIEW

File: src/assets/components/shared/DataGridTable/DataGridTable.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
+import { SxProps, Theme } from "@mui/material";
+import { DataGrid, GridColDef } from "@mui/x-data-grid";
+import { dataGridTableStyle } from "./styles";
+
+interface DataGridTableProps {
+  columns: GridColDef[];
+  rows: { [key: string]: any }[];
```
Comment (kfir-azu) [inline src/assets/components/shared/DataGridTable/DataGridTable.tsx:7]: That's a good place to use generics. 

```
export type TableRow<T> {
[key:string]: T
}
```

What do you think about that?

File: src/assets/components/shared/DataGridTable/DataGridTable.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+  searchTerm?: string;
```
Comment (kfir-azu) [inline src/assets/components/shared/DataGridTable/DataGridTable.tsx:8]: Is the search is part of the table itself?
I think its a different component that is not related to the table

File: src/assets/components/shared/DataGridTable/DataGridTable.tsx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  sortBy = { field: "id", sort: "asc" },
+  isLoading,
+  isError,
+  errorMessage,
+  rowsPerPage = 30,
+  pageSizeOptions = [rowsPerPage],
+  style,
+  onRowClick,
+}: DataGridTableProps) {
+  const filteredRows = searchTerm
```
Comment (kfir-azu) [inline src/assets/components/shared/DataGridTable/DataGridTable.tsx:32]: This logic should be at the parent component.
You will search in some page.
the generic dataGrid table only need to get the result of the filtered rows as the rows props and show it.

File: src/assets/components/shared/DataGridTable/DataGridTable.tsx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        )
+      )
+    : rows;
+
+  const handleRowClick = (row: { [key: string]: any }) => {
+    if (onRowClick) {
+      onRowClick(row);
+    }
+  };
+  const updatedColumns = columns.map((column) => ({
```
Comment (kfir-azu) [inline src/assets/components/shared/DataGridTable/DataGridTable.tsx:45]: Why do you need this functionality?

File: src/assets/components/shared/DataGridTable/DataGridTable.tsx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    ...column,
+    sortable: false,
+  }));
+
+  return (
+    <>
+      {isError ? (
+        <div>{errorMessage || "An error occurred while loading data."}</div>
+      ) : (
+        <DataGrid
```
Comment (kfir-azu) [inline src/assets/components/shared/DataGridTable/DataGridTable.tsx:55]: Please check about:
onSortModelChange
onPaginationModelChange
rowsPerPageOptions

File: src/assets/components/shared/DataGridTable/DataGridTable.tsx [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+          loading={isLoading}
+          onRowClick={(params) => handleRowClick(params.row)}
+          initialState={{
+            pagination: { paginationModel: { pageSize: rowsPerPage } },
+            sorting: {
+              sortModel: [sortBy],
+            },
+          }}
+          pageSizeOptions={pageSizeOptions}
+          filterModel={{
```
Comment (kfir-azu) [inline src/assets/components/shared/DataGridTable/DataGridTable.tsx:70]: The filter shouldn't be handled inside the component

File: src/assets/pages/AllPokemonsPage/AllPokemonsPage.tsx [hunk 7]
```diff
@@ -1,7 +1,31 @@
...
+
+//tests 
+const columns = [
+  { field: "id", headerName: "ID", width: 100 },
+  { field: "name", headerName: "Name", width: 200 },
+  { field: "description", headerName: "Description", width: 300 },
+];
+
+const rows = [
+  { id: 1, name: "Item 1", description: "Description of Item 1" },
```
Comment (kfir-azu) [inline src/assets/pages/AllPokemonsPage/AllPokemonsPage.tsx:11]: I Would also test what happens if you insert long text.
How does it look if you don't insert description - do you have empty state for a field in the table?
What happens if you don't send any data to the table? is it considered as an error or you show empty state?

**Review by kfir-azu — CHANGES_REQUESTED** [review]
Good job!

File: src/assets/components/shared/DataGridTable/DataGridTable.tsx (hunk 4 above)
Comment (Ran-Regev-HLS) [reply src/assets/components/shared/DataGridTable/DataGridTable.tsx:42]: just for styling purposes (having the arrow indicating sorting without it being functional)

File: src/assets/pages/AllPokemonsPage/AllPokemonsPage.tsx (hunk 7 above)
Comment (Ran-Regev-HLS) [reply src/assets/pages/AllPokemonsPage/AllPokemonsPage.tsx:11]: should this handled by generic table or by indevidual col/rows definition? 
im gonna have a type called PokemonTableColumns (or something like this) and there i assumed i will handles all this (styling/empty state/overflow)

File: src/assets/components/shared/Dropdown/Dropdown.tsx [hunk 8]
```diff
@@ -1,57 +1,60 @@
...
-        onChange(setlectedItem);
-      }
-    } else {
-      onChange(value);
-    }
-  };
-
+  style,
+}: DropdownProps) {
+  function renderItem(props: HTMLAttributes<HTMLLIElement>, option: itemType) {
```
Comment (kfir-azu) [inline src/assets/components/shared/Dropdown/Dropdown.tsx:32]:
```suggestion
  function renderItem(props: HTMLAttributes<HTMLLIElement>, option: OptionType) {
```

File: src/assets/components/shared/Dropdown/Dropdown.tsx [hunk 9]
```diff
@@ -1,57 +1,60 @@
...
+    return (
+      <li {...props} key={option.id}>
+        <Box sx={itemContainerStyles}>
```
Comment (kfir-azu) [inline src/assets/components/shared/Dropdown/Dropdown.tsx:35]: optionContainerStyles

File: src/assets/components/shared/Dropdown/Dropdown.tsx [hunk 10]
```diff
@@ -1,57 +1,60 @@
...
+  }
 
   return (
     <Autocomplete
       disabled={isDisabled}
       options={items}
-      getOptionLabel={getItemLabel}
-      onChange={(_event, value) => handleChange(value)}
-      sx = {dropdownStyles}
+      getOptionLabel={(item) => item.label}
```
Comment (kfir-azu) [inline src/assets/components/shared/Dropdown/Dropdown.tsx:50]:
```suggestion
      getOptionLabel={(option) => option.label}
```

**Review by kfir-azu — APPROVED** [review]

**Pull request closed without merging**